slyd/slyd/resources/annotations.py

```diff
@@ -44,61 +45,58 @@ def update_annotation(manager, spider_id, sample_id, annotation_id,
                       attributes):
     relationships = _load_relationships(attributes['data'])
     data = attributes['data'].get('attributes', {})
-    annotation_id, attribute = _split_annotation_id(annotation_id)
+    anno_id, _id = _split_annotation_id(annotation_id)
     annotation, sample = _get_annotation(manager, spider_id, sample_id,
-                                         annotation_id)
-    # Check if attribute->field mappings and required fields have been updated
-    if 'attribute' in data and attribute != data['attribute']:
-        field_id = annotation['annotations'].pop(attribute, None)
-        if relationships.get('field_id'):
-            field_id = relationships['field_id']
-        annotation['annotations'][data['attribute']] = field_id
-        required = set(annotation['required']) - {attribute}
-        if data.get('required'):
-            required += {data['attribute']}
-        else:
-            required -= {data['attribute']}
-        data['required'] = list(required)
-    elif data.get('required') and attribute in annotation['annotations']:
-        data['required'] = annotation['required'] + [attribute]
-    new_attribute = data.pop('attribute', attribute)
-    data.pop('required', None)
+                                         anno_id)
+    annotation_data = annotation['data'][_id]
+    field_id = annotation_data['field']
+    if relationships.get('field_id'):
+        field_id = relationships['field_id']
+    annotation_data['field'] = field_id
+    attribute = annotation_data['attribute']
+    if data.get('attribute'):
+        attribute = data['attribute']
+    annotation_data['attribute'] = attribute
```

Comment:
An attribute should always exist otherwise the annotation is useless since it won't extract any data.